File: chatbot/ChatMessage.c
//
//  ChatMessage.c
//  chatbot
//
//  Created by Jonathan Keller on 5/4/16.
//  Copyright © 2016 NobodyNada. All rights reserved.
//

#include "ChatMessage.h"
#include <stdlib.h>
#include <string.h>

ChatMessage *createChatMessage(const char *text, unsigned long messageID, ChatUser *user) {
    ChatMessage *m = malloc(sizeof(ChatMessage));
    m->content = malloc(strlen(text) + 1);
    strcpy(m->content, text);
    m->id = messageID;
    m->user = user;
    return m;
}

void deleteChatMessage(ChatMessage *m) {
    free(m->content);
    free(m);
}

File: FireAlarm/ChatMessage.swift
//
//  ChatMessage.swift
//  FireAlarm
//
//  Created by NobodyNada on 8/28/16.
//  Copyright © 2016 NobodyNada. All rights reserved.
//

import Foundation

open class ChatMessage {
	open let room: ChatRoom
    open let user: ChatUser
    open let content: String
    open let id: Int?
	open let replyID: Int?
	
	public init(room: ChatRoom, user: ChatUser, content: String, id: Int?, replyID: Int? = nil) {
		self.room = room
        self.user = user
        self.content = content
        self.id = id
		self.replyID = replyID
    }
}
